superset/views/core.py

```diff
@@ -965,28 +996,27 @@ def filter(self, datasource_type, datasource_id, column):
         return json_success(payload)
 
     @staticmethod
-    def remove_extra_filters(filters):
+    def remove_extra_filters(filters: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
         """Extra filters are ones inherited from the dashboard's temporary context
         Those should not be saved when saving the chart"""
         return [f for f in filters if not f.get("isExtra")]
 
     def save_or_overwrite_slice(
         self,
-        args,
-        slc,
-        slice_add_perm,
-        slice_overwrite_perm,
-        slice_download_perm,
-        datasource_id,
-        datasource_type,
-        datasource_name,
-    ):
+        slc: Slice,
+        slice_add_perm: bool,
+        slice_overwrite_perm: bool,
+        slice_download_perm: bool,
+        datasource_id: int,
+        datasource_type: str,
+        datasource_name: str,
+    ) -> FlaskResponse:
         """Save or overwrite a slice"""
-        slice_name = args.get("slice_name")
-        action = args.get("action")
+        slice_name = request.args.get("slice_name")
```

Comment:
No need to parse `args` which was merely `request.args` when `request` is global.